### PR TITLE
[Marketplace Contribution] Picus NG Content Pack

### PR DESCRIPTION
**Pull request by erdemkm**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Publish new content pack.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+        message = 'ok'
+    except Exception as e:
+        if 'Forbidden' in str(e) or 'Authorization' in str(e) or 'NewConnectionError' in str(e) or 'Unauthorized' in str(e) or picus_accessToken is None:
+            message = 'Authorization Error: make sure API Key or Picus URL is correctly set'
+        else:
+            raise e
+    return message
+
+def get_agent_list_command(client: Client) -> CommandResults:
+    picus_endpoint = f"/v1/agents"
```
Comment (ilaner) [inline Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG.py:91]: Please add this as a function in the client

File: Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG.py [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+    for agent in picus_agents:
+        agent["created_at"] = str(datetime.fromtimestamp(agent["created_at"] / 1000))
+
+    table_name = "Picus Agent List"
+    table_headers = ['id','name','status','type','version','created_at','platform_name','platform_architecture']
+    md_table = tableToMarkdown(table_name, picus_agents, headers=table_headers, removeNull=True, headerTransform=string_to_table_header)
+    return CommandResults(readable_output=md_table, outputs=picus_agents, outputs_prefix="Picus.agentlist")
+
+def get_agent_detail_command(client: Client) -> CommandResults:
+    picus_endpoint = "/v1/agents/"
```
Comment (ilaner) [inline Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG.py:104]: please add it as a function in the client class

File: Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG.py [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+    threat_ids = demisto.args().get('threat_ids')
+    threat_ids = list(threat_ids.split(","))
+    picus_action_raw_results = ""
+    picus_action_results: Dict[str, Any] = {"results": []}
+
+    offset = demisto.args().get('offset')
+    limit = demisto.args().get('limit')
+
+    if offset is not None and limit is None:
+        return_error("limit should be set.")
```
Comment (ilaner) [inline Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG.py:343]: raise DemistoException instead

File: Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG.py [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+    elif offset is None and limit is not None:
+        return_error("offset sohuld be set.")
```
Comment (ilaner) [inline Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG.py:345]: DemistoException

**Review by ilaner — CHANGES_REQUESTED** [review]
Thank you for your contribution. It looks much better
Please add functions in the Client for each endpoint (which recieves the url_suffix and the arguments and calls `http_request`). Use this functions in the `command` functions.

In addition, do not use `return_error` inside functions, just raise DemistoException. This will be caught in the main

**Review by ilaner — CHANGES_REQUESTED** [review]
No need to provide the client function the url_suffix, the url_suffix should specified inside any of those functions.

File: Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG_test.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
+import json
+import io
+
+# flake8: noqa
```
Comment (ilaner) [inline Packs/PicusNGAutomation/Integrations/PicusNG/PicusNG_test.py:4]: please remove this comment

File: package-lock.json [hunk 6]
```diff
@@ -66,11 +66,11 @@
       }
     },
     "node_modules/@babel/generator": {
-      "version": "7.18.12",
```
Comment (JudahSchwartz) [inline package-lock.json:69]: is there a reason this file is being changed? Can we revert changes here?

Comment (erdemkm) [reply package-lock.json:69]: I have no idea. I didn't change. if needed how can we fix it?

**Review by ilaner — APPROVED** [review]
Good job!

**Pull request merged**